metricflow/model/semantics/metric_semantics.py

```diff
@@ -1,35 +1,31 @@
 import logging
-
-from typing import Dict, List, FrozenSet, Set, Tuple, Sequence
+from typing import Dict, List, FrozenSet, Tuple, Sequence
 
 from metricflow.errors.errors import MetricNotFoundError, DuplicateMetricError, NonExistentMeasureError
 from metricflow.model.objects.metric import Metric, MetricType
 from metricflow.model.objects.user_configured_model import UserConfiguredModel
+from metricflow.model.semantics.data_source_join_evaluator import MAX_JOIN_HOPS
 from metricflow.model.semantics.data_source_semantics import DataSourceSemantics
-from metricflow.model.semantics.linkable_spec_resolver import ValidLinkableSpecResolver
 from metricflow.model.semantics.linkable_element_properties import LinkableElementProperties
+from metricflow.model.semantics.linkable_spec_resolver import ValidLinkableSpecResolver
 from metricflow.model.spec_converters import WhereConstraintConverter
+from metricflow.protocols.semantics import MetricSemanticsAccessor
 from metricflow.references import MetricReference
 from metricflow.specs import MetricSpec, LinkableInstanceSpec, MetricInputMeasureSpec, MeasureSpec
-from metricflow.model.semantics.data_source_join_evaluator import MAX_JOIN_HOPS
-
 
 logger = logging.getLogger(__name__)
 
 
-class MetricSemantics:  # noqa: D
+class MetricSemantics(MetricSemanticsAccessor):  # noqa: D
```

Comment:
Same thing about protocol inheritance from within a concrete implementation - seems it's best avoided